src/catalogue.js

```diff
@@ -92,17 +77,21 @@ export function loadItems(projectId, options = {}) {
         ...REQUEST_DEFAULTS,
         ...options,
         ...getRequiredDataSets(options)
-    }, ['dataSetIdentifier', 'returnAllDateDataSets']);
-
-    let bucketItems = get(cloneDeep(options), 'bucketItems.buckets');
-    if (bucketItems) {
-        bucketItems = bucketItemsToExecConfig(bucketItems);
-        return loadCatalog(
-            projectId,
-            {
+    }, [
+        'dataSetIdentifier',
+        'returnAllDateDataSets',
+        'attributesMap'
+    ]);
+
+    const mdObj = get(cloneDeep(options), 'bucketItems');
+    const { attributesMap } = options;
+    const hasBuckets = get(mdObj, 'buckets') !== undefined;
+    if (hasBuckets) {
+        return bucketItemsToExecConfig(projectId, mdObj, { attributesMap }).then(bucketItems =>
```

Comment:
[Q] Why is the `attributesMap` acquired defensively via `get` below and here it's not?